core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -46,7 +46,7 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
   /** Create a SparkConf that loads defaults from system properties and the classpath */
   def this() = this(true)
 
-  private[spark] val settings = new HashMap[String, String]()
+  private[spark] val settings = new HashMap[String, String] with SynchronizedMap[String, String]
```

Comment:
Yeah I think this is overall fairly low rate of access (e.g. we shouldn't be checking confs for every record) so seems reasonable to just use a built in.